worker/worker.py

```diff
@@ -215,6 +219,84 @@ def safe_sleep(f):
         print("\nSleep interrupted...")
 
 
+def text_hash(file):
+    with open(file) as f:  # text mode to have newline translation!
+        return base64.b64encode(
+            hashlib.sha384(f.read().encode("utf8")).digest()
+        ).decode("utf8")
+
+
+def generate_sri(install_dir):
+    sri = {}
+    for file in FILE_LIST:
+        item = os.path.join(install_dir, file)
+        sri[file] = text_hash(item)
+    return sri
+
+
+def write_sri(install_dir):
+    sri = generate_sri(install_dir)
+    sri_file = os.path.join(install_dir, "sri.txt")
+    print("Writing sri hashes to {}".format(sri_file))
+    with open(sri_file, "w") as f:
+        json.dump(sri, f)
+        f.write("\n")
+
+
+def verify_sri(install_dir):
+    sri = generate_sri(install_dir)
+    sri_file = os.path.join(install_dir, "sri.txt")
+    if not os.path.exists(sri_file):
+        print("{} does not exist".format(sri_file))
+        return False
+    try:
+        with open(sri_file, "r") as f:
+            sri_ = json.load(f)
+    except:
```

Comment:
Better to write the expected exceptions or to print them if unknown:
```python
except Exceptions as e:
    print("Exception reading {}:\n".format(sri_file), e, sep="", file=sys.stderr)
```

Reply:
Changed it!